django_extensions/management/commands/print_settings.py

```diff
@@ -26,34 +31,47 @@ def add_arguments(self, parser):
         )
         parser.add_argument(
             '--format',
-            default='simple',
+            default=DEFAULT_FORMAT,
             dest='format',
             help='Specifies output format.'
         )
         parser.add_argument(
             '--indent',
-            default=4,
+            default=DEFAULT_INDENT,
             dest='indent',
             type=int,
             help='Specifies indent level for JSON and YAML'
         )
+        parser.add_argument(
+            '--show-secrets',
+            default=DEFAULT_SECRETS,
+            dest='show secrets',
+            type=bool,
+            help='Specifies if should be revealed the value of secrets'
+        )
 
     @signalcommand
     def handle(self, *args, **options):
+
+        show_secrets, output_format, indent = self.get_defaults(options)
+
         a_dict = {}
 
-        for attr in dir(settings):
-            if self.include_attr(attr, options['setting']):
-                value = getattr(settings, attr)
-                a_dict[attr] = value
+        if show_secrets:
```

Comment:
We should be able to do this cleaner.... it smells like a lot of code duplication to me...

Makes more sense to do something like:

```python
if show_secrets:
    settings_dct = {k: getattr(settings, k) for k in dir(settings) if k.isupper()}
else:
    settings_dct = get_safe_settings()
```

Reply:
Sure, I agree!